### PR TITLE
Adds custom squads

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request
Still kinda WIP but good for testing

Basically Sls can create squads with a hex color code + name and set it's desc after, marines can join it using a console near the end of prep. They get their own squad channels as well.
Actual squad counts unchanged for now.

## Why It's Good For The Game
Lets ppl organize squads by squad role better

## Changelog
:cl:
add: Added custom squads;
add: SLs may now create their own squads using a terminal in/near their prep, and marines can join said squads using the squad selection consoles near preps
admin: admins can now delete squads
/:cl:

### CODE REVIEW

**Comment by Xander3359** [comment]
Alright boys, it's time to play as XanderBlue Squad. Anyone willing to join? We're gonna push caves and die like the TGMC intended as a group of chads.

**Comment by bluezorua** [comment]
this is either going to never be used or used exclusively to shitpost

**Comment by FosterGrime-Heymynameslouie** [comment]
jumpers

**Comment by XSlayer300** [comment]
Add the console to the Canterbury big and small variants? And maybe Arachnae once it is merged?

**Comment by TiviPlus** [comment]
dont wanna add it to crash yet

File: code/datums/jobs/squads.dm
Comment (Lumipharon) [inline code/datums/jobs/squads.dm:427]: the tracking id stuff is set in New and requires the squad name be already set.
I know you manually call this proc later, but init_squad will still init a squad which is probably not ideal. Not the end of the world tho.

File: code/game/objects/machinery/squad_manager.dm
Comment (Lumipharon) [inline code/game/objects/machinery/squad_manager.dm:27]: I assume you're not enabling this in HvH, but this prevents SOM from using it even though you've made the code support it.
```suggestion
	if(!ismarineleaderjob(user.job) && !issommarineleaderjob(user.job))
```

File: code/modules/admin/admin_verbs.dm
Comment (Lumipharon) [inline code/modules/admin/admin_verbs.dm:1218]: autodoc

**Review by Lumipharon — CHANGES_REQUESTED** [review]
Minor feedback. I'm going to just assume the tgui is good lol.
@RipGrayson for mapping changes though

**Review by RipGrayson — CHANGES_REQUESTED** [review]
Fine except for the fact that touches Minerva, the new consoles should be added to Arachne instead since Minerva doesn't exist.

File: code/modules/admin/admin_verbs.dm
Comment (TiviPlus) [reply code/modules/admin/admin_verbs.dm:1218]: done

File: code/datums/jobs/squads.dm
Comment (TiviPlus) [reply code/datums/jobs/squads.dm]: made it pass to new instead like color

**Review by RipGrayson — APPROVED** [review]

**Comment by Lumipharon** [comment]
pr is fine, just needs to pull from master for the funny hard del bug

**Comment by verywhiterabit** [comment]
Apparently, roles that usually get all squad-specific radio channels are missing the ability to hear radios from the custom squads.

**Comment by verywhiterabit** [comment]
also apparently custom squad SLs cant fire OB or railgun

**Comment by TiviPlus** [comment]
> also apparently custom squad SLs cant fire OB or railgun

Cant find anything that could cause that are you sure

**Comment by TiviPlus** [comment]
Also this should be good to go I'll make some followups after

**Comment by GrayRachnid** [comment]
> Apparently, roles that usually get all squad-specific radio channels are missing the ability to hear radios from the custom squads.

was this addressed

File: code/datums/jobs/squads.dm
Comment (vvvv-vvvv) [inline code/datums/jobs/squads.dm:454]: because only tgmc is guaranteed to exist
```suggestion
	LAZYADDASSOCSIMPLE(SSjob.active_squads, new_squad.faction, new_squad)
```

**Comment by GrayRachnid** [comment]
Isssues:

- Command staff still don't have access to custom radio channels
- The main overwatch console cannot view the Squad Monitor contents of custom squads

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**
